operator/deploy/crd.yaml

```diff
@@ -30,11 +30,335 @@ spec:
     listKind: KialiList
     plural: kialis
     singular: kiali
+  preserveUnknownFields: false
   scope: Namespaced
-  subresources:
-    status: {}
-  version: v1alpha1
   versions:
-  - name: v1alpha1
-    served: true
-    storage: true
+    - name: v1alpha1
+      served: true
+      storage: true
+      subresources:
+        status: {}
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                installation_tag:
+                  type: string
+                istio_component_namespaces:
+                  type: object
+                  additionalProperties: true
+                istio_namespace:
+                  type: string
+                api:
+                  type: object
+                  properties:
+                    namespaces:
+                      type: object
+                      properties:
+                        exclude:
+                          type: array
+                          items:
+                            type: string
+                          default:
```

Comment:
somewhere (either here, or maybe in `operator/roles/kiali-deploy/default/main.yaml`) we need a comment to tell people "if you change this default, you also need to change the default found in default/main.yaml AND in the kiali_cr.yaml file"  (and whereever else we define defaults).

Question: do we even need to define the defaults here? Can we avoid this? Otherwise, this is going to be a maintenance problem because now we have to keep the defaults all in sync with this CRD as well as the operator's ansible defaults and the kiali_cr example yaml.

If we have to define the defaults here (or if it just is best practice to do so) then we should remove the notion that kiali_cr defines the defaults and instead in kiali_cr comments we should tell the reader to come here to the CRD to find out what the defaults are (this would eliminate the problem if needing to sync these files).

Reply:
> Question: do we even need to define the defaults here? Can we avoid this?
We don't need the defaults :)
I'll remove them. These are remnants of my first try.

I was putting all defaults I saw and I realized that it was going to be too much work to keep this in sync. It's better to omit the defaults and let the operator do its job of setting those values..